api/cost/cost-request-fn.ts

```diff
@@ -0,0 +1,47 @@
+import middy from "@middy/core";
+import { sqsClient } from "../../utils/aws-sdk/sqs";
+import { SQSEvent, SQSRecord } from "aws-lambda";
+import { logger } from "../../utils/logging";
+import { CostRequest, CspRequest } from "../../models/cost-jobs";
+import { SendMessageCommand } from "@aws-sdk/client-sqs";
+import { injectLambdaContext } from "@aws-lambda-powertools/logger";
+import inputOutputLogger from "@middy/input-output-logger";
+import errorLogger from "@middy/error-logger";
+import { cspRequest } from "../util/csp-request";
+import { errorHandlingMiddleware } from "../../utils/middleware/error-handling-middleware";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+
+const COST_RESPONSE_QUEUE_URL = process.env.COST_RESPONSE_QUEUE_URL ?? "";
+export const MESSAGE_GROUP_ID = "cost-response-queue-message-group";
+
+async function baseHandler(event: SQSEvent): Promise<void> {
+  const processedMessages = [];
+  if (event.Records) {
```

Comment:
```suggestion
  if (event.Records?.length) {
```

Should we also just skip this if `Records` is empty (in additional to undefined/falsey)?

Reply:
Yeah agreed.